geomesa-core/src/main/scala/org/locationtech/geomesa/core/index/AttributeIdxStrategy.scala

```diff
@@ -591,4 +461,214 @@ object AttributeIndexStrategy {
+        (prop, AccRange.exact(exact))
+
+      case f: PropertyIsLike =>
+        val prop = f.getExpression.asInstanceOf[PropertyName].getPropertyName
+        // Remove the trailing wildcard and create a range prefix
+        val literal = f.getLiteral
+        val value = if (literal.endsWith(QueryStrategyDecider.MULTICHAR_WILDCARD)) {
+          literal.substring(0, literal.length - QueryStrategyDecider.MULTICHAR_WILDCARD.length)
+        } else {
+          literal
+        }
+        (prop, AccRange.prefix(getEncodedAttrIdxRow(sft, prop, value)))
+
+      case _ =>
+        val msg = s"Unhandled filter type in attribute strategy: ${filter.getClass.getName}"
+        throw new RuntimeException(msg)
+    }
+
+  private def greaterThanRange(sft: SimpleFeatureType, prop: String, lit: AnyRef): AccRange = {
+    val rowIdPrefix = getTableSharingPrefix(sft)
+    val start = new Text(getEncodedAttrIdxRow(sft, prop, lit))
+    val endPrefix = AttributeTable.getAttributeIndexRowPrefix(rowIdPrefix, sft.getDescriptor(prop))
+    val end = AccRange.followingPrefix(new Text(endPrefix))
+    new AccRange(start, false, end, false)
+  }
+
+  private def greaterThanOrEqualRange(sft: SimpleFeatureType, prop: String, lit: AnyRef): AccRange = {
+    val rowIdPrefix = getTableSharingPrefix(sft)
+    val start = new Text(getEncodedAttrIdxRow(sft, prop, lit))
+    val endPrefix = AttributeTable.getAttributeIndexRowPrefix(rowIdPrefix, sft.getDescriptor(prop))
+    val end = AccRange.followingPrefix(new Text(endPrefix))
+    new AccRange(start, true, end, false)
+  }
+
+  private def lessThanRange(sft: SimpleFeatureType, prop: String, lit: AnyRef): AccRange = {
+    val rowIdPrefix = getTableSharingPrefix(sft)
+    val start = AttributeTable.getAttributeIndexRowPrefix(rowIdPrefix, sft.getDescriptor(prop))
+    val end = getEncodedAttrIdxRow(sft, prop, lit)
+    new AccRange(start, false, end, false)
+  }
+
+  private def lessThanOrEqualRange(sft: SimpleFeatureType, prop: String, lit: AnyRef): AccRange = {
+    val rowIdPrefix = getTableSharingPrefix(sft)
+    val start = AttributeTable.getAttributeIndexRowPrefix(rowIdPrefix, sft.getDescriptor(prop))
+    val end = getEncodedAttrIdxRow(sft, prop, lit)
+    new AccRange(start, false, end, true)
+  }
+
+  // This function assumes that the query's filter object is or has an attribute-idx-satisfiable
+  //  filter.  If not, you will get a None.get exception.
+  def partitionFilter(query: Query, sft: SimpleFeatureType): (Query, Filter) = {
+
+    val filter = query.getFilter
+
+    val (indexFilter: Option[Filter], cqlFilter) = filter match {
+      case and: And =>
+        findFirst(AttributeIndexStrategy.getAttributeIndexStrategy(_, sft).isDefined)(and.getChildren)
+      case f: Filter =>
+        (Some(f), Seq())
+    }
+
+    val nonIndexFilters = filterListAsAnd(cqlFilter).getOrElse(Filter.INCLUDE)
+
+    val newQuery = new Query(query)
+    newQuery.setFilter(nonIndexFilters)
+
+    if (indexFilter.isEmpty) {
```

Comment:
i'd probably prefer do a .getOrElse(throw new Exception...) and return that, than do a .get below.